instrumentation/google.golang.org/grpc/otelgrpc/test/grpc_test.go

```diff
@@ -411,7 +437,10 @@ func checkStreamServerSpans(t *testing.T, spans []trace.ReadOnlySpan) {
 		semconv.RPCService("grpc.testing.TestService"),
 		otelgrpc.RPCSystemGRPC,
 		otelgrpc.GRPCStatusCodeKey.Int64(int64(codes.OK)),
-	}, streamInput.Attributes())
+		semconv.NetSockPeerAddr("127.0.0.1"),
+	}, filterKVs(streamInput.Attributes(), semconv.NetSockPeerPortKey))
+	// Client port is random
+	assert.True(t, existKVs(streamInput.Attributes(), semconv.NetSockPeerAddrKey))
```

Comment:
How about creating a helper like

```go
func findAttribute(attr []attribute.KeyValue, key attribute.Key) attribute.KeyValue
	for _, a := range attr {
		if a.Key == key {
			return a
		}
	}
	return attribute.KeyValue{}
}
```

and use it like this:

```go
netSockPeerPort := findAttribute(streamInput.Attributes(), semconv.NetSockPeerPortKey)
require.NoZero(t, netSockPeerPort)
assert.ElementsMatch(t, []attribute.KeyValue{
		semconv.RPCMethod("StreamingInputCall"),
		semconv.RPCService("grpc.testing.TestService"),
		otelgrpc.RPCSystemGRPC,
		otelgrpc.GRPCStatusCodeKey.Int64(int64(codes.OK)),
		semconv.NetSockPeerAddr("127.0.0.1"),
		netSockPeerPort 
	}, streamInput.Attributes())
```

Reply:
Even this should work

```go
ssert.ElementsMatch(t, []attribute.KeyValue{
		semconv.RPCMethod("StreamingInputCall"),
		semconv.RPCService("grpc.testing.TestService"),
		otelgrpc.RPCSystemGRPC,
		otelgrpc.GRPCStatusCodeKey.Int64(int64(codes.OK)),
		semconv.NetSockPeerAddr("127.0.0.1"),
		findAttribute(streamInput.Attributes(), semconv.NetSockPeerPortKey),
	}, streamInput.Attributes())
```

Reply:
I don't think we would want to rely on the empty attribute.  There is a chance both the port attribute doesn't exist, returning the empty attribute, and the instrumentation puts an empty attribute in the list.  Together this test would pass.

If the call was `findAttribute() (attribute.KeyValue, bool)` that would work, but doesn't really simplify the tests.

Reply:
Then the first code snippet: https://github.com/open-telemetry/opentelemetry-go-contrib/pull/4503#discussion_r1380830445 would work for you thanks to 

```go
require.NoZero(t, netSockPeerPort)
```

Personally, I think that a case when the instrumentation puts an empty attribute in the list is such an edge case that I would not worry about it.